src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -365,18 +353,10 @@ func (cfg Config) fetchCloudConfig() (bytes []byte, err error) {
 func (cfg Config) fetchCloudConfigForAddr(proxyAddr string) ([]byte, error) {
 	log.Tracef("fetchCloudConfigForAddr '%s'", proxyAddr)
 
-	if proxyAddr != "" {
-		// Wait for proxy to become available
-		err := waitforserver.WaitForServer("tcp", proxyAddr, 30*time.Second)
-		if err != nil {
-			return nil, fmt.Errorf("Proxy never came up at %v: %v", proxyAddr, err)
-		}
-	}
+	dialer := cfg.Client.HighestQOSFrontedDialer()
```

Comment:
Two things:

 - The config URL needs be changed to plain HTTP instead of HTTPS (see [this comment](https://github.com/getlantern/lantern/blob/valencia/src/github.com/getlantern/fronted/dialer.go#L54)).  I'll change that.

`config/config.go` is shared between the client and the server.  In the server side, `dialer` becomes `nil` and we get a runtime panic trying to access it.  We could try to somehow create a fronted dialer in the server side.  I haven't checked how hard that would be, but anyway that seems silly since servers need no fronting.  We could somehow pass a `dialer` to this function but it gets hard to reason about whether the dialer will be in sync with the current config.

I think the easiest thing to do would be to have a cfg.GetDirectHttpClient() method like such (from the top of my head; I haven't checked it):

```go
func (cfg *Config) GetDirectHttpClient() (*http.Client, error) {
    if Config.Role == "server" {
        ret, err := util.HTTPClient("", "")
        if err != nil {
            return nil, fmt.Errorf("Couldn't create direct server-side HTTP client: %v", err)
        }
        return ret, nil
    } else {
        return cfg.Client.HighestQOSFrontedDialer().DirectHttpClient(), nil
    }
}        
```